### PR TITLE
runner/local ensure checkout release in runner, cleanup

### PR DESCRIPTION
**Pull request by dgassaway**
- runner releases checkout resource if run cancelled before checkout is complete
- some cleanup wrt multiple uses of "run" in simple.go
- runner final updateStatus calls clearer, not relying on inner function logic
- test package rename per convention
- set stdout/stderr refs in run status sooner, and preserve them if status updated later

Main issue was IDd via code inspection - thinking about how to implement positive/negative tests to verify checkout.Release() calls

### CODE REVIEW

File: runner/local/simple.go [hunk 1]
```diff
@@ -88,47 +79,57 @@ func (r *simpleRunner) StatusAll() ([]runner.ProcessStatus, error) {
...
 		// We are ending the running task.
 		// depend on the invariant that there is at most 1 run with !state.IsDone(),
 		// so if we're changing a Process from not Done to Done it must be running
 		log.Printf("local.simpleRunner: run done. %+v", new)
 		close(r.running.doneCh)
 		r.running = nil
 	}
+
+	r.runs[new.RunId] = new
 	return new, nil
```
Comment (jasonbs10) [inline runner/local/simple.go]: As part of this cleanup could you rename 'new' to something else?

File: runner/local/simple.go [hunk 2]
```diff
@@ -137,12 +138,16 @@ func (r *simpleRunner) run(cmd *runner.Command, runId runner.RunId, doneCh chan
 	// Wait for checkout or cancel
 	select {
 	case <-doneCh:
+		go func() {
```
Comment (jasonbs10) [inline runner/local/simple.go]: Now that I look at it, it seems like we would get a nil pointer exception if doneCh closes and 'defer checkout.Release()' gets called before checkout completes. The simplest solution would be to 'defer func() { if checkout != nil { checkout.Release() } }()

File: runner/local/simple.go (hunk 1 above)
Comment (dgassaway) [inline runner/local/simple.go]: Yep, done!

File: runner/local/simple.go (hunk 2 above)
Comment (dgassaway) [inline runner/local/simple.go]: I think I'm missing something :( As I'm reading it: if doneCh is closed, we start a goroutine that waits on checkoutDone and calls Release.  If doneCh is not closed, we just block until checkoutDone is closed and then defer the Release statement, so in both cases waiting for checkoutDone to be closed.

Comment (jasonbs10) [inline runner/local/simple.go]: Nope you're not missing anything, I misread.

File: runner/local/simple.go [hunk 3]
```diff
@@ -88,47 +79,57 @@ func (r *simpleRunner) StatusAll() ([]runner.ProcessStatus, error) {
 	return statuses, nil
 }
 
-func (r *simpleRunner) Abort(run runner.RunId) (runner.ProcessStatus, error) {
-	return r.updateStatus(runner.AbortStatus(run))
+func (r *simpleRunner) Abort(runId runner.RunId) (runner.ProcessStatus, error) {
```
Comment (dbentley) [inline runner/local/simple.go]: Ugh, we've done fooId everywhere up til now, but I think Go style says to do fooID. Not sure if we should be consistent, or start doing the right thing and go back and change at some point... WDYT?

File: runner/local/simple.go [hunk 4]
```diff
@@ -88,47 +79,57 @@ func (r *simpleRunner) StatusAll() ([]runner.ProcessStatus, error) {
...
-		return old, nil
+
+	if oldStatus.State.IsDone() {
+		return oldStatus, nil
 	}
-	r.runs[new.RunId] = new
-	if new.State.IsDone() {
+
+	if newStatus.State.IsDone() {
+		if oldStatus.StdoutRef != "" && newStatus.StdoutRef == "" {
```
Comment (dbentley) [inline runner/local/simple.go]: You can skip the oldStatus.StdoutRef != ""... same result, but it's less code...

File: runner/local/simple.go (hunk 2 above)
Comment (dbentley) [inline runner/local/simple.go]: But what about if checkout is nil? Won't this panic?

File: runner/local/simple.go (hunk 3 above)
Comment (dgassaway) [inline runner/local/simple.go]: Seems like the Go way is to prefer "ID" so I will go with that for files touched in this review.  If we want to revert files to sue that I can make a ticket, or we can start requesting changes to other reviews as it comes up.

File: runner/local/simple.go (hunk 2 above)
Comment (dgassaway) [inline runner/local/simple.go]: Going to add some nil checks

File: runner/local/simple.go (hunk 4 above)
Comment (dgassaway) [inline runner/local/simple.go]: will do

**Pull request merged**